crates/transpiler/src/angle_bound_registry.rs

```diff
@@ -0,0 +1,149 @@
+use pyo3::Python;
+
+use crate::TranspilerError;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::PhysicalQubit;
+
+#[derive(Clone)]
+pub(crate) enum CallbackType {
+    Python(PyObject),
+    Native(fn(&[f64], &[PhysicalQubit]) -> DAGCircuit),
+}
+
+impl CallbackType {
+    fn call(&self, angles: &[f64], qubits: &[PhysicalQubit]) -> PyResult<DAGCircuit> {
+        match self {
+            Self::Python(inner) => {
+                let qubits: Vec<usize> = qubits.iter().map(|x| x.index()).collect();
+                Python::with_gil(|py| inner.bind(py).call1((angles, qubits))?.extract())
+            }
+            Self::Native(inner) => Ok(inner(angles, qubits)),
+        }
+    }
+}
+
+/// Registry of Angle Wrapping function
+///
+///
+#[pyclass(module = "qiskit._accelerate.angle_bound_registry")]
+#[pyo3(name = "WrapAngleRegistry")]
+pub struct PyWrapAngleRegistry(WrapAngleRegistry);
+
+#[pymethods]
+impl PyWrapAngleRegistry {
+    #[new]
+    pub fn new() -> Self {
+        PyWrapAngleRegistry(WrapAngleRegistry::new())
+    }
+
+    /// Get a replacement circuit for
+    pub fn substitute_angle_bounds(
+        &self,
+        name: &str,
+        angles: Vec<f64>,
+        qubits: Vec<u32>,
+    ) -> PyResult<Option<DAGCircuit>> {
+        let qubits: Vec<PhysicalQubit> = qubits.into_iter().map(PhysicalQubit::new).collect();
+        self.0.substitute_angle_bounds(name, &angles, &qubits)
+    }
+
+    pub fn add_wrapper(&mut self, name: String, callback: PyObject) {
+        self.0.registry.insert(name, CallbackType::Python(callback));
+    }
+
+    fn __getstate__(&self, py: Python) -> PyResult<Py<PyDict>> {
+        let bounds_dict = PyDict::new(py);
+        for (name, bound) in self.get_inner().registry.iter() {
+            if let CallbackType::Python(obj) = bound {
+                bounds_dict.set_item(name, obj.clone_ref(py))?;
+            } else {
+                return Err(TranspilerError::new_err(
+                    "Target contains native code bounds callbacks which can't be serialized",
+                ));
+            }
+        }
+        Ok(bounds_dict.unbind())
+    }
+
+    fn __setstate__(&mut self, data: Bound<PyDict>) -> PyResult<()> {
+        for (key, val) in data.iter() {
+            let name: String = key.extract()?;
+            self.add_wrapper(name, val.unbind());
+        }
+        Ok(())
+    }
+}
+
+impl PyWrapAngleRegistry {
+    pub fn get_inner(&self) -> &WrapAngleRegistry {
+        &self.0
+    }
+}
+
+pub struct WrapAngleRegistry {
+    registry: HashMap<String, CallbackType>,
+}
+
+impl WrapAngleRegistry {
+    pub fn new() -> Self {
+        WrapAngleRegistry {
+            registry: HashMap::new(),
+        }
+    }
+
+    pub fn add_native(
+        &mut self,
+        name: String,
+        callback: fn(&[f64], &[PhysicalQubit]) -> DAGCircuit,
+    ) {
+        self.registry.insert(name, CallbackType::Native(callback));
+    }
+
+    /// Get a replacement circuit for
```

Comment:
Finish your docstrings 🙃

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14406/commits/f84003a9af4b9b4e1c907477f7f7e41049d80339